sources/models/src/modeled_types/kubernetes.rs

```diff
@@ -1002,6 +1006,177 @@ mod test_topology_manager_policy {
 
 // =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
 
+/// imageGCHighThresholdPercent is the percent of disk usage after which image
+/// garbage collection is always run. The percent is calculated by dividing this
+/// field value by 100, so this field must be between 0 and 100, inclusive. When
+/// specified, the value must be greater than imageGCLowThresholdPercent.
+/// Default: 85
+/// https://kubernetes.io/docs/reference/config-api/kubelet-config.v1beta1/
+
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct ImageGCHighThresholdPercent {
+    // All settings are brought in as Strings.
```

Comment:
This isn't accurate. Models certainly can hold non-string values; `max_pods` is an example.

This particular approach is used to add validation rules on top of a basic type, such as `String`. Historically we added this to enforce constraints on strings, like `SingleLineString`.

It looks like we've used `inner: String` for non-string values elsewhere, but that was probably a mistake. This probably should be `inner: i32` but then we'd need an `i32_impls_for!` macro to make the struct behave like an `i32` after validation.

I won't insist on doing that as part of this PR, but it might be good to backlog an issue for extending this validation framework to types besides `String`.

Reply:
I have opened issue #2257 and my updated changes will still use `inner: String` (until #2257 is implemented).